src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileRepository.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class FileRepository : IFileRepository
+    {
+        private readonly ContentDbContext _contentDbContext;
+
+        public FileRepository(ContentDbContext contentDbContext)
+        {
+            _contentDbContext = contentDbContext;
+        }
+
+        public async Task<ReleaseFileReference> CreateOrUpdate(string filename,
```

Comment:
This was originally lifted from the old `ReleaseFilesService.cs :643` `CreateOrUpdateFileReference`  now deleted.

The update is only used in one place, for updating a filename when a chart is replaced, and the create used when we create a zip file is really simple too, since we nevere have a source or a replacement. Therefore all I have done is reduced `CreateOrUpdate` into a `Create` and added a simpler version `CreateZip` and an `UpdateFilename`.

Reply:
Cool, fine by me :+1: